Sources/Core/Renderer/GUI/GUIElement/GUITextInput.swift

```diff
@@ -12,21 +13,46 @@ struct GUITextInput: GUIElement {
     ).meshes(context: context)
 
     // Message
-    var message = try content.meshes(context: context)
-    message.translate(amount: [2, 2])
+    var firstMessage = try String(content.prefix(upTo: content.index(content.endIndex, offsetBy: -cursorIndex))).meshes(context: context)
+    var secondMessage = try String(content.suffix(from: content.index(content.endIndex, offsetBy: -cursorIndex))).meshes(context: context)
+    if firstMessage.size().x != 0 {
+      firstMessage.translate(amount: [2, 2])
+      secondMessage.translate(amount: [3 + firstMessage.size().x, 2])
+    } else {
+      firstMessage.translate(amount: [0, 2])
+      secondMessage.translate(amount: [2, 2])
+    }
+    // When at the end of the message, use an underscore cursor
+    if cursorIndex == 0 {
+      var cursor: [GUIElementMesh] = []
+      if Int(CFAbsoluteTimeGetCurrent() * 10/3) % 2 == 1 {
+        cursor = try "_".meshes(context: context)
+        let messageWidth = firstMessage.size().x + secondMessage.size().x
+        var xOffset = 2
+        if messageWidth != 0 {
+          xOffset = messageWidth + 4
+        }
+        cursor.translate(amount: [xOffset, 2])
+      }
+      let message = firstMessage + secondMessage
+      return background + message + cursor
+    // Otherwise, use a vertical bar cursor
```

Comment:
Merge this comment into the one on line 25 so that it doesn't need to be weirdly indented.